### PR TITLE
Support `INTERVAL` SQL Type

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/5651

# Rationale for this change

See https://github.com/apache/arrow-datafusion/issues/5651 (basically I want to be able to interact with intervals like normal). Also related to https://github.com/apache/arrow-datafusion/pull/5764 from @berkaysynnada

# What changes are included in this PR?
1. Add basic support for `INTERVAL` SQL --> `DataType::Interval(MonthDayNano)` arrow type
2. Add code to go to/from ScalarValue and Interval Arrays
3. Basic tests

# Are these changes tested?
Yes

# Are there any user-facing changes?

Yes (you can now do `SELECT '1 day'::interval`) and it works!

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/interval.slt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# under the License.
+
+
+# Use `interval` SQL literal syntax
+query TT
+select
+  arrow_typeof(interval '5 months'),
+  arrow_typeof(interval '5 days 3 nanoseconds')
+----
+Interval(YearMonth) Interval(MonthDayNano)
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/interval.slt:25]: The fact that this is Interval YearMonth is an artifact -- I think `interval ..` syntax should always be the same type. I will file a ticket to that effect

File: datafusion/core/tests/sqllogictests/test_files/interval.slt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+query TT
+select
+  arrow_typeof(cast ('5 months' as interval)),
+  arrow_typeof(cast ('5 days 3 nanoseconds' as interval))
+----
+Interval(MonthDayNano) Interval(MonthDayNano)
+
+
+# Create tables with interval values
+# (TODO file ticket)
```
Comment (tustvold) [inline datafusion/core/tests/sqllogictests/test_files/interval.slt:46]: ?

**Review by tustvold — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/interval.slt (hunk 2 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/interval.slt:46]: Will file a ticket. Thank you for the review

File: datafusion/core/tests/sqllogictests/test_files/interval.slt (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/interval.slt:26]: https://github.com/apache/arrow-datafusion/issues/5801

File: datafusion/core/tests/sqllogictests/test_files/interval.slt (hunk 2 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/interval.slt]: https://github.com/apache/arrow-datafusion/issues/5802

**Pull request merged**